### PR TITLE
Connect to external processes

### PR DESCRIPTION
**Pull request by MikeInnes**
This adds a simple (but much nicer than before) UI for connecting from an external Julia process, e.g. in the terminal or IJulia.

### CODE REVIEW

File: lib/connection/messages.coffee [hunk 1]
```diff
@@ -75,3 +76,19 @@ module.exports =
       else
         ""
       dismissable: true
+
+  connectExternal: ->
+    tcp.listen().then (port) ->
+      msg = atom.notifications.addInfo "Connect an external process",
+        detail: """
+        To connect a Julia process running in the terminal,
+        run the command:
```
Comment (pfitzseb) [inline lib/connection/messages.coffee]: Maybe mention that the current process needs to be stopped first? Or re-add the `direquire` stuff below?

File: lib/connection/messages.coffee [hunk 2]
```diff
@@ -75,3 +76,19 @@ module.exports =
...
+
+            using Juno
+            Juno.connect(#{port})
+        """
+        dismissable: true
```
Comment (pfitzseb) [inline lib/connection/messages.coffee]: What would you think about adding a button that will copy the necessary commands to the clipboard? Shouldn't be too hard, see [here](https://atom.io/docs/api/v1.10.2/NotificationManager#instance-addInfo).

File: lib/connection/messages.coffee (hunk 1 above)
Comment (MikeInnes) [inline lib/connection/messages.coffee]: That's my oversight, fixed

File: lib/connection/messages.coffee (hunk 2 above)
Comment (MikeInnes) [inline lib/connection/messages.coffee]: Great idea, I've added this. It's something we can think about applying to our other buttons as well.

**Comment by MikeInnes** [comment]
Thanks for the great feedback!

**Comment by pfitzseb** [comment]
Anytime -- it's great to have this be so much more convenient!

**Comment by Balinus** [comment]
Is there a how-to about how someine can connect to a remote julia session (through SSH)?

**Comment by pfitzseb** [comment]
You can forward the local port used by Atom to the remote machine, see https://github.com/JunoLab/atom-julia-client/issues/220#issuecomment-250208410.

**Comment by MikeInnes** [comment]
@Balinus If you need more instruction on this, feel free to open an issue on JunoDocs.jl and we can make time to document it in more detail.

**Comment by Balinus** [comment]
Thanks, my first attempt didn't work. Here's what I did : 

1- Connect to the remote server through ssh with this command : "ssh -L localport:localhost:localport2 server.domain.com"
2- launch Julia on the remote server. Then launch command 
"import Atom"
"@async Atom.serve(localport2)" --> Got "juno-msg-readyTask (runnable) @0x00007fa0ccdce4a0"
3- on my local machine in Atom/Juno : "Julia Client: connect to local port" --> I'm asked to enter the port. I enter "localport"once I enter the port the julia local process is started but is not connected to the remote machine.

I will put this message in JunoDocs.jl. Thanks!

Both machines are on a LAN.

**Comment by Balinus** [comment]
I've finally succeeded connecting to an external julia session on a remote machine. Here's my steps 

```
1. Launch Atom/Juno locally
2. Julia-client : connect to external process (Juno will then tell you which port to use locally --> we'll call it **PORT#2**)
3. Connect to remote server : "**ssh -R PORT#1:localhost:PORT#2 your.server.com**"
4. Launch julia in the terminal
5. type in julia session : "**using Juno; Juno.connect(PORT#1)**"
```
Hope this helps anyone trying to do that. It's a really nice feature (Atom on our server is sluggish and almost unusable).

**Comment by Balinus** [comment]
Me again... :)

When Atom is connected to an external process, is there any way to define the backend gui for plots? Right now, I can't use any plotting functions.

```
julia> using Plots
WARNING: Method definition show(IO, Base.Multimedia.MIME{:text/plain}, Plots.Plot{#B<:Any}) in module Plots at /home/proy/.julia/v0.5/Plots/src/output.jl:170 overwritten in module Main at /home/proy/.julia/v0.5/Plots/src/output.jl:263.

julia> plot([1:10])
WARNING: No working GUI backend found for matplotlib.
/home/proy/.julia/v0.5/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/font_manager.py:1288: UserWarning: findfont: Font family [u'Helvetica'] not found. Falling back to Bitstream Vera Sans
  (prop.get_family(), self.defaultFamily[fontext]))
/home/proy/.julia/v0.5/Conda/deps/usr/lib/python2.7/site-packages/matplotlib/figure.py:397: UserWarning: matplotlib is currently using a non-GUI backend, so cannot show the figure
  "matplotlib is currently using a non-GUI backend, "
```

**Comment by AsafManela** [comment]
Thanks for the feature and how-to!

**Comment by merl-dev** [comment]
> in Atom/Juno : Julia-client : connect to external process (Juno will then tell you which port to use locally --> we'll call it PORT#2)

Not sure what this refers to, since there is no such option through the menu, nor any such setting in the julia-client settings.

**Comment by Balinus** [comment]
On my computer (Linux) I use the following shortcut for the command palette `ctrl-shift-p`. Then, you can write the command `external process`, which should give you the correct you.

**Pull request merged**